extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -533,6 +533,24 @@ public void setRoleClaimSeparator(String roleClaimSeparator) {
         @ConfigItem
         public Optional<String> cookiePath = Optional.empty();
 
+        /**
+         * If this property is set to 'true' then a normal 302 redirect response will be returned. If
+         * the request was initiated via an XMLHttpRequest, then that redirect will automatically be
+         * followed which may not be desiravle for single page applications.
+         * If it is false, a status code of '444' will be returned to allow the client to handle the
+         * redirect manually.
+         */
+        @ConfigItem(defaultValue = "true")
+        public boolean autoRedirect;
```

Comment:
@boosey IMHO this property name should be clear it is about XHR, otherwise one may get an impression that if `it is auto-redirect=false` then we don't redirect in the code flow, but we are only doing an XHR specific exception here,
As such I suggest to rename it something similar to what @pedroigor proposed in the other discussion, `redirect-with-xhr`, something like that. And update the docs:
```
* If it is false, a status code of '...' will be returned to allow the client to handle the redirect manually only if client uses XHR....
```

Reply:
I have committed a working solution that I have been able to verify with a sample client. Actually, I was doing it all wrong. It is not the redirect exception that needs to be updated, but the CHALLENGE_DATA when it is an XHR and autoRedirect=false.

I didn't see your comments about 401 vs 399 until now, so that is not changed yet. Neither did I see the comment about the setting name which I will also change in the next day or so. 

I've revoked the changes to the AuthenitcationRedirectException except for the getCode method where I actually changed it to return the actual member variable value instead of a hard coded 302.

Reply:
@boosey Good catch, I forgot why I added in the first place :-)